lib/config-generator.js

```diff
@@ -325,6 +321,49 @@ class ConfigGenerator {
             .map((plugin) => plugin.plugin);
     }
 
+    buildOptimizationConfig() {
+        const optimization = {
+
+        };
+
+        if (this.webpackConfig.isProduction()) {
+            optimization.minimizer = [
+                uglifyPluginUtil(this.webpackConfig)
+            ];
+        } else {
+            // see versioning.js: this gives us friendly module names,
+            // which can be useful for debugging, especially with HMR
+            optimization.namedModules = true;
+        }
+
+        let splitChunks = {};
+        if (this.webpackConfig.sharedCommonsEntryName) {
+            const cacheGroups = {};
+            cacheGroups[this.webpackConfig.sharedCommonsEntryName] = {
+                chunks: 'initial',
+                name: this.webpackConfig.sharedCommonsEntryName,
+                // important: setting this to the entry name means that
+                // all modules included by that entry go into this cache group
+                test: this.webpackConfig.sharedCommonsEntryName,
+                // seems to default the rest of the options (like minSize)
+                // to settings so that modules matched by "test" are
+                // *definitely* included.
+                enforce: true,
+            };
+
+            splitChunks.cacheGroups = cacheGroups;
+            // causes a runtime.js to be emitted with the Webpack runtime
+            optimization.runtimeChunk = 'single';
```

Comment:
shouldn't this be done by checking whether `optimization.splitChunks` is a non-empty object instead ? Now that you added support for a callback, the case of using `sharedCommonsEntryName` is not the only case triggering splitting.